juju/juju
# Wires up user correlation for models in DQlite.

Until now we haven't had any strong referential integrity for model owners onto the user table. Now that the user DDL has been established this sets up the checks and test to assert that when adding a model the owner exists and is a valid user in the system (not removed).

As part of this PR I have also wired up the bootstrap process to set the admin user as the owner of the controller model.

## Checklist


## QA steps

Run unit tests under `service` and `state` in the model domain package.

Also bootstrap a controller to lxd and make sure it succeeds with no error output around creating the controller model.

## Documentation changes

N/A

## Links

**Jira card:** JUJU-5345
